security/pkg/stsservice/tokenmanager/tokenmanager.go

```diff
@@ -112,3 +112,8 @@ func (tm *TokenManager) DumpTokenStatus() ([]byte, error) {
 func (tm *TokenManager) SetPlugin(p Plugin) {
 	tm.plugin = p
 }
+
+// GetPlugin gets token exchange plugin
+func (tm *TokenManager) GetPlugin() Plugin {
```

Comment:
I don't think we should be exposing the implementation details like this. Either it should be an interface or it shouldn't be - but treating it as both a generic interface then casting it later is a bit unusual

Reply:
Also please stop using such broad terms as 'plugin' - what kind of plugin ? It's enough to have 'the proxy protocol' craziness.

Reply:
GetPlugin() has been removed.